### ISSUE TITLE
dotnet add package fails even when an unreachable source is disabled in Nuget.config

### ISSUE THREAD

**Issue by kcwms** [dotnet/cli#7491]
The "Unreachable" source is valid when connected to my corporate network. In the 1.* version of the tool I'm certain that I would simply disable the **mycorporatenuget** source and the dotnet add package command worked as expected ... or maybe I just imaged that.

## Work Around ##
manually edit %appdata%\NuGet\Nuget.Config and remove or comment out the line that includes the offending source

## Steps to reproduce
1. dotnet new console in an empty directory
1. dotnet add package log4net -s https://api.nuget.org/v3/index.json

## Expected  behavior
`  <ItemGroup>
    <PackageReference Include="log4net" Version="2.0.8" />
  </ItemGroup>` in *.csproj file

## Actual behavior
```
  Writing C:\Users\<me>\AppData\Local\Temp\tmp5E49.tmp
info : Adding PackageReference for package 'log4net' into project 'C:\Users\<me>\Documents\projects\nugetfail\nugetfail.csproj'.
log  : Restoring packages for C:\Users\<me>\Documents\projects\nugetfail\nugetfail.csproj...
info :   GET http://mycorporatenuget/nuget/FindPackagesById()?id='log4net'
info :   CACHE https://api.nuget.org/v3-flatcontainer/log4net/index.json
info :   GET https://dotnetmyget.blob.core.windows.net/artifacts/dotnet-core/nuget/v3/flatcontainer/log4net/index.json
info :   NotFound https://dotnetmyget.blob.core.windows.net/artifacts/dotnet-core/nuget/v3/flatcontainer/log4net/index.json 463ms
log  : Retrying 'FindPackagesByIdAsyncCore' for source 'http://mycorporatenuget/nuget/FindPackagesById()?id='log4net''.
log  : An error occurred while sending the request.
log  :   The server name or address could not be resolved
info :   GET http://mycorporatenuget/nuget/FindPackagesById()?id='log4net'
log  : Retrying 'FindPackagesByIdAsyncCore' for source 'http://mycorporatenuget/nuget/FindPackagesById()?id='log4net''.
log  : An error occurred while sending the request.
log  :   The server name or address could not be resolved
info :   GET http://mycorporatenuget/nuget/FindPackagesById()?id='log4net'
error: Failed to retrieve information about 'log4net' from remote source 'http://mycorporatenuget/nuget/FindPackagesById()?id='log4net''.
error:   An error occurred while sending the request.
error:   The server name or address could not be resolved

```

## Environment data
`dotnet --info` output:
.NET Command Line Tools (2.0.0)

Product Information:
 Version:            2.0.0
 Commit SHA-1 hash:  cdcd1928c9

Runtime Environment:
 OS Name:     Windows
 OS Version:  10.0.10240
 OS Platform: Windows
 RID:         win10-x64
 Base Path:   C:\Program Files\dotnet\sdk\2.0.0\

Microsoft .NET Core Shared Framework Host
  Version  : 2.0.0
  Build    : e8b8861ac7faf042c87a5c2f9f2d04c98b69f28d

`content of %appdata%\NuGet\Nuget.Config`
``` 
<?xml version="1.0" encoding="utf-8"?>
<configuration>
  <packageSources>
    <add key="nuget.org" value="https://api.nuget.org/v3/index.json" protocolVersion="3" />
    <add key="mycorporatenuget" value="http://mycorporatenuget/nuget" />
  </packageSources>
  <disabledPackageSources>
    <add key="Microsoft and .NET" value="true" />
    <add key="mycorporatenuget" value="true" />
  </disabledPackageSources>
  <packageRestore>
    <add key="enabled" value="True" />
    <add key="automatic" value="True" />
  </packageRestore>
  <bindingRedirects>
    <add key="skip" value="False" />
  </bindingRedirects>
</configuration>
```

**Comment by akanieski** [comment]
Ah. This is a problem for me as well.. some of my sources are behind a VPN for some of my projects.. but it breaks dotnet add while working on other non VPN'd projects.. .. even after you disable a source.

**Comment by livarcocc** [comment]
This issue was moved to NuGet/Home#5867

**Comment by akanieski** [comment]
Not sure this belongs with nuget. It seems this only comes up with dotnet client. Nuget client has a flag to ignore failed sources, I don't see that flag for dotnet client.
		

		Get Outlook for iOS
	





On Mon, Sep 11, 2017 at 12:09 AM -0400, "Livar" <notifications@github.com> wrote:












Closed #7491.

**Comment by livarcocc** [comment]
The implementation of dotnet add package is owned by Nuget, which is why I moved the issue there.
